orc8r/cloud/go/obsidian/handler.go

```diff
@@ -174,15 +174,27 @@ func AttachHandlers(e *echo.Echo, handlers []Handler, m ...echo.MiddlewareFunc)
 // HttpError wraps the passed error as an HTTP error.
 // Code is optional, defaulting to http.StatusInternalServerError (500).
 func HttpError(err error, code ...int) *echo.HTTPError {
-	var status = http.StatusInternalServerError
-	if len(code) > 0 && code[0] >= http.StatusContinue &&
-		code[0] <= http.StatusNetworkAuthenticationRequired {
+	status := http.StatusInternalServerError
+	if len(code) > 0 && isValidResponseCode(code[0]) {
 		status = code[0]
 	}
-	glog.Infof("REST HTTP Error: %s, Status: %d", err, status)
+	// TODO(hcgatewood): we should be handling REST error logging and metrics via middleware
+	if isServerErrCode(status) {
+		glog.Infof("REST HTTP Error: %s, Status: %d", err, status)
+	} else {
+		glog.V(1).Infof("REST HTTP Error: %s, Status: %d", err, status)
+	}
 	return echo.NewHTTPError(status, grpc.ErrorDesc(err))
 }
 
+func isServerErrCode(code int) bool {
+	return code >= http.StatusInternalServerError && code <= http.StatusNetworkAuthenticationRequired
+}
+
+func isValidResponseCode(code int) bool {
+	return code >= http.StatusContinue && code <= http.StatusNetworkAuthenticationRequired
+}
+
```

Comment:
This refactoring is twice longer & I think, harder to read. And the logic is redundant. Why is it needed?

A 3 line change should accomplish all that you are trying to do:

` func HttpError(err error, code ...int) *echo.HTTPError {
	var (
		status = http.StatusInternalServerError
		l glog.Level
	)
	if len(code) > 0 && code[0] >= http.StatusContinue &&
		code[0] <= http.StatusNetworkAuthenticationRequired {
		status = code[0]
		if status < http.StatusInternalServerError {
			l = 1
		}
	}
	glog.V(l).Infof("REST HTTP Error: %s, Status: %d", err, status)
	return echo.NewHTTPError(status, grpc.ErrorDesc(err))
}
`

Reply:
The main reason for this refactor is to, by default, only log errors if they're 5xx

In terms of implementation: creating named functions is one way to communicate the intent behind the if statements. Another option would have been to use Go's compound if statements

Reply:
glog.V(l).Infof(...) int the example, I gave above is using variable "l" as in "level"...

Reply:
Ah I see yes totally missed that, saw it as a `1`